ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
+ */
+static inline uintptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
 {
-  return (*((intptr_t *) &name) << 1) | 0x1;
+    uintptr_t tmp, sentinel = 0x1;
```

Comment:
Just wondering: does every compiler understand that 0x1 means "put a 1 in the left-most bit" as opposed to "set the value of this variable to 1"?